kernels/portable/cpu/op_upsample_nearest2d.cpp

```diff
@@ -79,11 +104,25 @@ Tensor& upsample_nearest2d_vec_out(
   const auto kernel_scale_w = area_pixel_compute_scale<double>(
       in.sizes()[3], out.sizes()[3], false, scale_w);
 
+  if (tensor_is_default_dim_order(in)) {
```

Comment:
I don't think there is a need to add this if/else here.
Rename the existing `upsample_nearest2d_kernel_impl` as `upsample_nearest2d_kernel_default` or something like that, and then move this if/else login inside a new `upsample_nearest2d_kernel_impl`.

Reply:
Not as far as I know. I don't think we have that machinery developed. As a matter of fact, I believe this is the first portable op branching out on dim order.
The few ops that currently support channel last in the portable library, are not dispatching specific code for that. In those cases the code is written in a dim order agnostic way, maybe only requiring that all inputs have the same dim order